### PR TITLE
Fixed #34286 -- Fixed admindocs markups for case-sensitive template/view names.

### PR DESCRIPTION
**Pull request by skidipap**
ticket from https://code.djangoproject.com/ticket/34286

add conditional logic for :view: and prevent href reference change to lowercase

### CODE REVIEW

**Review by felixxm — COMMENTED** [review]
@skidipap Thanks :+1: Please add a regression test.

**Comment by skidipap** [comment]
add simple test for :view:

**Comment by skidipap** [comment]
> @skidipap Thanks +1 It seems that `:template:` is also affected, e.g.:
> 
>  simplescreenrecorder-2023-02-02_08.11.34.mp4

i will fix it soon

**Comment by skidipap** [comment]
> @skidipap Thanks +1 It seems that `:template:` is also affected, e.g.:
> 
>  simplescreenrecorder-2023-02-02_08.11.34.mp4

i will fix it soon

> > @skidipap Thanks +1 It seems that `:template:` is also affected, e.g.:
> > simplescreenrecorder-2023-02-02_08.11.34.mp4
> 
> i will fix it soon

fix for :view: and :template:

File: django/contrib/admindocs/utils.py [hunk 1]
```diff
@@ -104,14 +104,17 @@ def create_reference_role(rolename, urlbase):
...
+        text_ref = text if name in exclude_name else text.lower()
         node = docutils.nodes.reference(
             rawtext,
             text,
             refuri=(
                 urlbase
                 % (
                     inliner.document.settings.link_base,
-                    text.lower(),
+                    text_ref,
```
Comment (felixxm) [inline django/contrib/admindocs/utils.py:117]: We know the `rolename` before, so checking `name` for each call seems like a waste. Can we do this in advance? E.g.

```diff
diff --git a/django/contrib/admindocs/utils.py b/django/contrib/admindocs/utils.py
index 6edff502ec..30df96ded0 100644
--- a/django/contrib/admindocs/utils.py
+++ b/django/contrib/admindocs/utils.py
@@ -101,6 +101,8 @@ ROLES = {
 
 
 def create_reference_role(rolename, urlbase):
+    is_case_sensitive = rolename in ["view", "template"]
+
     def _role(name, rawtext, text, lineno, inliner, options=None, content=None):
         if options is None:
             options = {}
@@ -111,7 +113,7 @@ def create_reference_role(rolename, urlbase):
                 urlbase
                 % (
                     inliner.document.settings.link_base,
-                    text.lower(),
+                    text if is_case_sensitive else text.lower(),
                 )
             ),
             **options,
```

Comment (skidipap) [reply django/contrib/admindocs/utils.py:117]: sure, nice

**Comment by felixxm** [comment]
@skidipap Thanks :+1: Welcome aboard :boat: 

I pushed small edits.

**Comment by skidipap** [comment]
> @skidipap Thanks +1 Welcome aboard boat
> 
> I pushed small edits.

thank you

> @skidipap Thanks +1 Welcome aboard boat
> 
> I pushed small edits.

u too, thank u for maintaining this framework

**Pull request merged**